lib/experimental/block-comments.php

```diff
@@ -26,6 +26,58 @@ function gutenberg_block_comment_add_post_type_support() {
 }
 add_action( 'init', 'gutenberg_block_comment_add_post_type_support' );
 
+/**
+ * Register comment metadata for block comment status.
+ */
+function gutenberg_register_block_comment_metadata() {
+	register_meta(
+		'comment',
+		'_wp_block_comment_status',
+		array(
+			'type'              => 'string',
+			'description'       => __( 'Block comment resolution status', 'gutenberg' ),
+			'single'            => true,
+			'show_in_rest'      => true,
+			'auth_callback'     => function () {
+				return current_user_can( 'edit_posts' );
+			},
+			'sanitize_callback' => function ( $value ) {
+				return in_array( $value, array( 'resolved', 'reopen' ), true ) ? $value : '';
+			},
```

Comment:
```suggestion
			'type'          => 'string',
			'description'   => __( 'Block comment resolution status', 'gutenberg' ),
			'single'        => true,
			'show_in_rest'  => array(
				'schema' => array(
					'type' => 'string',
					'enum' => array( 'resolved', 'reopen' ),
				),
			),
			'auth_callback' => function () {
				return current_user_can( 'edit_posts' );
			},
```

I think it's easier to define a schema than to perform sanitization.